crypto/evp/evp_lib.c

```diff
@@ -303,8 +303,10 @@ int EVP_Cipher(EVP_CIPHER_CTX *ctx, unsigned char *out,
             return
                 ctx->cipher->ccipher(ctx->provctx, out, &outl,
                                      inl + (blocksize == 1 ? 0 : blocksize),
-                                     in, (size_t)inl);
-        return 0;
+                                     in, (size_t)inl)
+                ? (int)outl
+                : -1;
+        return -1;
```

Comment:
Note that pre-3.0 `EVP_Cipher()` doesn't check if `ctx->cipher->do_cipher` is NULL or not.  We should probably add an error message here...